### PR TITLE
Fetching and Instantiating User through data model

### PR DESCRIPTION
**Pull request by JWoltjen**
#### What does  this PR do?

- debugs code in several different files so that webpack can successfully compile and run on localHost8080. 
- builds Fetch.js, which handles the asynchronous get fetches, and returns them using Promise.all
- builds domUpdates.js, which acts as an empty receptacle to be filled by the currentUser once it is fetched from the API. 
- builds getData() function on scripts page which uses the fetched data to instantiate a currentUser out of the UserRepository
- Allows developer to access random currentUser with class properties with and data on window load. 

#### Where should the reviewer start?

- the reviewer should start by looking at fetch.js to see how the data is retrieved. 
- the reviewer should then continue to scripts.js to see how the retreived data is synthesized and instantiated
- the reviewer should finish at domUpdates.js to see how the instantiated data is then distributed for use within the DOM. 

#### How should this be manually tested?

The PR has been manually tested on this developer's machine and it is currently running on localHost8080 without errors. After reviewing the rest of the PR, this can be verified by running npm start on the reviewer's local machine after the pull request has been merged. 

#### Any background context you want to provide?

The old code uses "user" as its variable for an instantiated member of the UserRepository class, while our program uses "currentUser". Because of this, in order to demonstrate MVP--an instantiated random user on load--almost all of the original code in scripts.js relating to display has been commented out for later refactoring.

### CODE REVIEW

File: dist/index.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <html>
   <head>
     <meta charset='UTF-8'>
-    <title>Trisha & Kayla's FitLit</title>
+    <title>Demaceo, Peyton and Jeff's FitLit</title>
```
Comment (peytonjo) [inline dist/index.html:5]: Doesn't look like too much was changed here

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
 		"webpack-dev-server": "^3.1.14"
 	},
 	"dependencies": {
-		"jquery": "^3.4.1"
+		"jquery": "^3.4.1",
+		"moment": "^2.29.1"
```
Comment (peytonjo) [inline package.json:36]: not quite sure what "moment" is but Im sure this is tracking the current date

File: src/User.js [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
      this.name = userData.name;
      this.address = userData.address;
      this.email = userData.email;
      this.strideLength = userData.strideLength;
      this.dailyStepGoal = userData.dailyStepGoal;
      this.friends = userData.friends;
-     this.hydrationInfo = new HydrationRepository(date);
-     this.activityInfo = new ActivityRepository(date);
-     this.sleepInfo = new SleepRepository(date);
+     this.hydrationInfo = new HydrationRepository(todaysDate);
```
Comment (peytonjo) [inline src/User.js:14]: Looks good glad we are getting that date functionality down

File: src/UserRepository.js [hunk 4]
```diff
@@ -6,12 +6,14 @@ class UserRepository {
...
-  linkdata(rawData, todaysDate){
-    let instantiatedUsers = rawData.userData.map((rawUser) => new User(rawUser, todaysDate)
-      this.linkHydration(instantiedUsers, rawData.hydrationData);
-      this.linkSleep(instantiatedUsers, rawData.sleepData)
+  linkData(rawData, todaysDate){
+    let instantiatedUsers = rawData.userData.map(
+      (rawUser) => new User(rawUser, todaysDate)
+    );
+      this.linkHydration(instantiatedUsers, rawData.hydrationData);
+      this.linkSleep(instantiatedUsers, rawData.sleepData);
```
Comment (peytonjo) [inline src/UserRepository.js:14]: Great job with the  dynamic date functionality

File: src/domupdates.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const domUpdates = {
```
Comment (peytonjo) [inline src/domupdates.js:1]: domUpdates looks very new I look forward to getting into this topic more

File: src/fetch.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      return data.activityData;
+  })
+    .catch(err => console.log(err.message))
+
+  return Promise.all([userData, hydrationData, sleepData, activityData])
+    .then(data => {
+        let allData = {}
+        allData.userData = data[0];
+        allData.hydrationData = data[1];
+        allData.sleepData = data[2];
```
Comment (peytonjo) [inline src/fetch.js:36]: glad we got all of these methods moved over!

**Review by peytonjo — APPROVED** [review]
Looks good!

File: src/fetch.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    .catch(err => console.log(err.message));
+
+  let activityData = fetch("https://fe-apps.herokuapp.com/api/v1/fitlit/1908/activity/activityData")
+    .then(response => response.json())
+    .then(data => {
+      return data.activityData;
+  })
+    .catch(err => console.log(err.message))
+
+  return Promise.all([userData, hydrationData, sleepData, activityData])
```
Comment (demaceo) [inline src/fetch.js:31]: Like this succinct way of consolidating all of the fetch data into this `Promise.all `

File: dist/index.html (hunk 1 above)
Comment (demaceo) [reply dist/index.html:5]: it is now **OUR** FitLit site

File: package.json (hunk 2 above)
Comment (demaceo) [reply package.json:36]: ^ ditto.

Comment (demaceo) [reply package.json:36]: `moment` is a package that helps format the current date.

File: src/domupdates.js (hunk 5 above)
Comment (demaceo) [reply src/domupdates.js:1]: ^ agreed! definitely will need a 101 on this file

File: src/fetch.js (hunk 6 above)
Comment (demaceo) [reply src/fetch.js:36]: yeah! 😅

**Pull request merged**